File: frontend/quiz/src/seleniumTests/Pages/LogInPage.js
import Page from "./Page";
import { getElementById } from "./Utility";
const usernameId = "username";
const passwordId = "password";
const loginBtnId = "login";
const singBtnId = "signUp";
const enterAsGuestBtnId = "enterAsGuest";
class LoginPage extends Page {
  get enterAsGuestButton() {
    return getElementById(this.driver, enterAsGuestBtnId);
  }

  get logInBtn() {
    return getElementById(this.driver, loginBtnId);
  }

  async open() {
    await super.open("http://localhost:3000/");
  }
  get signUpBtn() {
    return getElementById(this.driver, singBtnId);
  }
  get passwordInput() {
    return getElementById(this.driver, passwordId);
  }
  get usernameInput() {
    return getElementById(this.driver, usernameId);
  }
}

export default LoginPage;

File: backend/selenium/LogInPage.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.expected_conditions import presence_of_element_located
from Utility import getElementById, isVisibleById
from Page import Page
usernameId = "username"
passwordId = "password"
loginBtnId = "login"
singBtnId = "signUp"
enterAsGuestBtnId = "enterAsGuest"


class LogInPage(Page):
    def enterAsGuestButton(self):
        return getElementById(self.driver, enterAsGuestBtnId)

    def logInBtn(self):
        return getElementById(self.driver, loginBtnId)

    def signUpBtn(self):
        return getElementById(self.driver, singBtnId)

    def passwordInput(self):
        return getElementById(self.driver, passwordId)

    def usernameInput(self):
        return getElementById(self.driver, usernameId)
